src/mpid/ch4/shm/posix/posix_rma.h

```diff
@@ -168,14 +168,14 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_do_put(const void *origin_addr,
             mpi_errno =
                 MPIR_Ilocalcopy_gpu(origin_addr, origin_count, origin_datatype, 0, &origin_attr,
                                     target_addr, target_count, target_datatype, 0, &target_attr,
-                                    MPL_GPU_COPY_D2D_OUTGOING, engine_type, 1, &yreq);
+                                    MPL_GPU_COPY_DIRECTION_NONE, engine_type, 1, &yreq);
```

Comment:
Does `_NONE` supersede other types since it is "smart"?

Reply:
Can you add document comment in `src/mpl/include/mpl_gpu.h` on the semantics of copy direction enums. In particular, we need not that `_NONE` is supposed to work for all directions. This is not implemented in CUDA or HIP yet, right? Thus the notes on the semantics are important for additional implementations.

Reply:
Added some documentation on `_D2D_OUTGOING`, `_D2D_INCOMING`, and `_NONE`. Let me know if any of these need expanded comments